e2e/app.e2e.spec.ts

```diff
@@ -45,6 +47,8 @@ describe('iam-cache-server E2E tests', () => {
       .useValue(await spawnIpfsDaemon())
       .overrideProvider(Provider)
       .useValue(provider)
+      .overrideProvider(ConfigService)
+      .useValue(configService)
```

Comment:
Could you please give more details why to disable validation in tests? For me validation is passing without overriding ConfigService here and having `DID_REGISTRY_ADDRESSS` in `.env`

Reply:
Nevermind, I can see `"DID_REGISTRY_ADDRESS" is required`. This supposdly means that only values from `.env` are validated. May be we can use custom validation function https://docs.nestjs.com/techniques/configuration#custom-validate-function, which use values from both process and .env?

Reply:
So `DID_REGISTRY_ADDRESS` isn't found, because validation happens at the time, when test file imports `app.module` https://github.com/energywebfoundation/ssi-hub/blob/b8eaa5f3f29048c7c1640664fc41da78a30ff00d/e2e/app.e2e.spec.ts#L8 instead of when application is created. I think this is a good reason to move ConfigModule.forRoot in @Module

Reply:
It behaves exactly the same when moved to the `@module`. I've checked this already. `ConfigModule.forRoot` is executed before the decorator.

Reply:
The `ConfigService` does this when instantiated, so here we have the `process.env` object populated with them, but the problem was caused by validation added to the ConfigService. As a result, ConfigService creates its own copy of the `process.env` values before `process.env.DID_REGISTRY_ADDRESS = didRegistry.address;` is executed so manipulating them here leads to inconsistencies. This is the reason why providing another instance of it helped.

I already tried to provide a custom `validate` function to `ConfigModule.forRoot` but it behaves the same.